src/app/controllers/public-forms.server.controller.js

```diff
@@ -44,9 +44,13 @@ exports.isFormPublic = function (req, res, next) {
  * @param  {Object} res - Express response object
  */
 exports.redirect = async function (req, res) {
+  const queryPortion = req.url ? req.url.split('?', 2)[1] : undefined
   let redirectPath = req.params.state
     ? req.params.Id + '/' + req.params.state
     : req.params.Id
+  if (queryPortion && queryPortion.length > 0) {
+    redirectPath = redirectPath + '?' + encodeURIComponent(queryPortion)
+  }
   try {
     const metaTags = await fetchMetatags(req)
     return res.render('index', {
```

Comment:
Not related to this PR, but how does this code work? How does redirect work when we already return a render before the redirect?

Reply:
Self answered, the final redirect is for when metatags fail to be created... should probably move it into the catch block with a return statement for clarity

Reply:
@karrui can file an issue for this? this PR needs to go out quickly

Reply:
This doesn't affect functionality, all g. I'll fix it in my refactor

Reply:
But I want to check, comments point to metatags only being created if it's a crawler, but I don't see any code checking if it's a crawler? Seems like all redirects create the metatags and render index regardless? Is it old code or?

Reply:
yeah, this is pretty old code that Kevan hastily wrote when he was briefly on forms.

Reply:
done